### DISCUSSION TITLE
Laravel Nova Resource Index Query not respecting ordering when scout is enabled

### DISCUSSION THREAD

**Discussion by oluomotoso** [laravel/nova-issues#5731] [Q&A]
- Laravel Version:10.8.0
- Nova Version: 4.23.0
- PHP Version: 8.2
- Browser type and version: #.#
- Reproduction Repository: https://github.com/###/###

### Description:

When Laravel Scout is enabled - the resource index field is not respecting the default sort

### Detailed steps to reproduce the issue on a fresh Nova installation:
Install and configure Laravel Nova with Laravel Scout enabled.
Create a resource in Laravel Nova with a default sorting defined in the indexQuery method.
Observe that the default sorting is not applied to the index results.

**Comment by crynobone** [comment]
This is the expected behaviour so Scout can utilise the relevant search results instead of just depending on `indexQuery()`. You should use `scoutQuery()` and interacts with `Laravel\Scout\Builder` to change the behaviour.

**Reply by oluomotoso** [reply]
This is not when a search is done. It happens even on the default index on resource load, is that expected too?

**Reply by crynobone** [reply]
Yes, that is the expected behavior even in Orion releases.

**Reply by oluomotoso** [reply]
```
 public static $defaultSort = ['created_at' => 'desc'];
 public static function scoutQuery(NovaRequest $request, $query)
    {
        return $query->orderBy(key(static::$defaultSort), reset(static::$defaultSort));
    }
```

This is not working too! There's nothing I do to sort that's working .

**Reply by crynobone** [reply]
It is impossible to change the default ordering when Scout searching is enabled except when search query is used and it's possible to use `scoutQuery()`.

**Reply by oluomotoso** [reply]
And you believe this is best practise? We have big transactions table that needs to be searched using Scout. At the same time , we need to always see the latest data on page load. Both are very important - with this it means I have to select one of the two.  Is there anyway I can amend the base query to change this?

**Reply by crynobone** [reply]
Scout Searching still would use `indexQuery()` during search and allow ordering to be used with remove any search-relevant result ordering from Meilisearch, Algolia etc, this is why we discourage having any `orderBy()` to be used within `indexQuery()` for resources using Scout.

You can override `Laravel\Nova\Query\Builder` using Laravel Container.

**Reply by oluomotoso** [reply]
Please , could you give an example of how I can achieve this?

**Reply by crynobone** [reply]
Alternatively you can only change the menu to include the default ordering (sortable field)

```php
    /**
     * Return the menu item that should represent the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return \Laravel\Nova\Menu\MenuItem
     */
    public function menu(Request $request)
    {
        return parent::menu($request)->data([
            'users_order' => 'email',
            'users_direction' => 'desc',
        ]);
    }
```

**Reply by oluomotoso** [reply]
Thanks alot for this. this works seemlessly on the index but doesn't work on relationships like hasmany etc. Is there any way to make that work too?
